experiment/coverage_utils.py

```diff
@@ -165,9 +196,39 @@ def generate_coverage_report(self):
             return
 
         src_dir = self.report_dir
-        dst_dir = exp_path.filestore(self.report_dir)
+        dst_dir = exp_path.filestore(src_dir)
         filestore_utils.cp(src_dir, dst_dir, recursive=True, parallel=True)
 
+    def generate_segment_function_csv(self):
+        """Stores the segment and function info in CSV file"""
+        segment_df_column_names = ["benchmark", "fuzzer", "trial_id",
+                                   "filename", "line", "col", "hits"]
+        segment_df = pd.DataFrame(columns=segment_df_column_names)
+
+        function_df_column_names = ["benchmark", "fuzzer", "trial_id",
+                                    "function_name", "hits"]
+        function_df = pd.DataFrame(columns=function_df_column_names)
+
+        for trial_id in self.trial_ids:
+            seg_df, func_df = \
+                extract_covered_segments_and_functions_from_summary_json(
+                    os.path.join(self.benchmark_fuzzer_measurement_dir,
+                                 'coverage_summary_{0}.json'.format(trial_id)),
+                    benchmark=self.benchmark,
+                    fuzzer=self.fuzzer,
+                    trial_id=trial_id)
+            segment_df = pd.concat([segment_df, seg_df], axis=0,
+                                   ignore_index=True)
+            function_df = pd.concat([function_df, func_df], axis=0,
+                                    ignore_index=True)
+
+        segment_csv_src = os.path.join(self.benchmark_fuzzer_measurement_dir,
+                                       'segments.csv')
+        function_csv_src = os.path.join(self.benchmark_fuzzer_measurement_dir,
+                                        'functions.csv')
```

Comment:
This PR is about generating two CSV files containing coverage information on segments and functions.
* The format of segment.csv is `benchmark, fuzzer, trial_id, filename, line, col, hits`
* The format of functions.csv is `benchmark, fuzzer, trial_id, function_name, hits`

We currently have one file for *each* (benchmark, fuzzer) in `benchmark_fuzzer_dir`, but eventually we would like to have one file for *all* benchmarks and fuzzers in the parent directory of `benchmark_fuzzer_dir`. 

We currently don't have time stamps. Any suggestions to add those are welcome.

Reply:
If file size is a concern, we could move names (benchmark, fuzzer, file_name, function_name) into a separate file and reference them by ID.